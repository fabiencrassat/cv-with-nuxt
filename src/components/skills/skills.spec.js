import Skills from './skills';
import testUtils from '~~/lib/tests/testUtils';

const stubs = ['SvgIcon'];

describe('skills', () => {
  it('mounts properly', () => {
    expect.hasAssertions();
    testUtils.mountProperly(Skills);
  });

  it('renders properly', () => {
    expect.hasAssertions();
    expect(testUtils.htmlFactory(Skills)).toMatchInlineSnapshot('<div></div>');
  });
  it('renders properly with props', () => {
    expect.hasAssertions();
    expect(
      testUtils.htmlFactory(Skills, {
        stubs,
        propsData: {
          skills: {
            foo: {
              label: 'foo',
              svg: 'SvgIcon',
              items: [
                {
                  percentage: 90,
                  label: 'item at 90%',
                },
                {
                  percentage: 60,
                  label: 'item at 60%',
                },
              ],
            },
            bar: {
              label: 'bar',
              svg: 'SvgIcon',
              items: [
                {
                  percentage: 50,
                  label: 'item at 50%',
                },
              ],
            },
          },
        },
      })
      // eslint-disable-next-line jest/prefer-inline-snapshots
    ).toMatchSnapshot();
  });
});
