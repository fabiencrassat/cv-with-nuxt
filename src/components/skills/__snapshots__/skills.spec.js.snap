// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skills renders properly with props 1`] = `
<div>
  <div class="flex bg-white shadow-lg rounded-lg md:mx-auto max-w-md md:max-w-2xl mb-4">
    <div class="flex items-start px-4 py-6 w-full">
      <svgicon-stub class="w-12 mr-4"></svgicon-stub>
      <div class="w-full">
        <h2 class="text-lg font-semibold text-gray-900 -mt-1 mb-4">
          foo
        </h2>
        <progressbar-stub label="item at 90%" percentage="90" class="mb-4"></progressbar-stub>
        <progressbar-stub label="item at 60%" percentage="60" class="mb-4"></progressbar-stub>
      </div>
    </div>
  </div>
  <div class="flex bg-white shadow-lg rounded-lg md:mx-auto max-w-md md:max-w-2xl mb-4">
    <div class="flex items-start px-4 py-6 w-full">
      <svgicon-stub class="w-12 mr-4"></svgicon-stub>
      <div class="w-full">
        <h2 class="text-lg font-semibold text-gray-900 -mt-1 mb-4">
          bar
        </h2>
        <progressbar-stub label="item at 50%" percentage="50" class="mb-4"></progressbar-stub>
      </div>
    </div>
  </div>
</div>
`;
